src/gui/tray/activitydata.cpp

```diff
@@ -93,10 +112,17 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
                 parameterJsonObject.value(QStringLiteral("id")).toString(),
                 parameterJsonObject.value(QStringLiteral("name")).toString(),
                 parameterJsonObject.contains(QStringLiteral("path")) ? parameterJsonObject.value(QStringLiteral("path")).toString() : QString(),
-                parameterJsonObject.contains(QStringLiteral("link")) ? QUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
+                stringToUrl(parameterJsonObject.value(QStringLiteral("link")).toString()),
             };
         }
 
+        if (const auto event = QStringLiteral("event");
+            activity._objectType == QStringLiteral("calendar") &&
+            activity._link.isEmpty() &&
+            activity._subjectRichParameters.contains(event)) {
+            activity._link = activity._subjectRichParameters[event].link;
```

Comment:
maybe an easier way to define the `"event"` constant string would be to use
```
namespace {
constexpr auto eventC = "event";
}
```
at the top of the file